modules/server_new/src/main/scala/observe/server/ObserveEngineImpl.scala

```diff
@@ -1478,13 +1487,49 @@ class ObserveEngineImpl[F[_]: Async: Logger](
     enabled:  SubsystemEnabled,
     clientId: ClientId
   ): F[Unit] =
-    toggleOverride("DHS",
-                   (enabled, x) => if (enabled.value) x.enableDhs else x.disableDhs,
-                   SetDhsEnabled(obsId, user.some, enabled),
-                   obsId,
-                   user,
-                   enabled,
-                   clientId: ClientId
+    toggleOverride(
+      "DHS",
+      (enabled, x) => if (enabled.value) x.enableDhs else x.disableDhs,
+      SetDhsEnabled(obsId, user.some, enabled),
+      obsId,
+      user,
+      enabled,
+      clientId: ClientId
     )
 
+  // Reloads the current atom if the observation is edited in ODB, only if the sequence is not running.
+  // We want this to reload regenerated steps if the QA of their recent execution does not pass.
+  // If the sequence is running, this signal is ignored and the atom will be reloaded at the end of current step anyway.
+  private def processObsEditOdbSignal(obsId: Observation.Id): F[Unit] =
+    Logger[F].debug(s"Observation [$obsId] was edited in ODB") >>
+      executeEngine
+        .offer:
+          Event.modifyState:
+            executeEngine.get
+              .map(EngineState.atSequence(obsId).getOption(_))
+              .flatMap: seq =>
+                if seq.exists(x => !x.seq.status.isRunning) then
```

Comment:
What if we are in pause?

Reply:
If the exposure is paused, the sequence state is still `Running`, so it's OK.

If the sequence is paused, it's safe to refresh the sequence.

Reply:
I see